### DISCUSSION TITLE
The alignment of the elements inside the <p:row> element is broken

### DISCUSSION THREAD

**Discussion by goxostar** [primefaces/community#1156] [PrimeFaces] [answered]
Hi,

I don't know if this is the right place to get support, but I have a problem.

I switched from Primefaces version 8.0 to version 10.0. After the switch, the alignment of the objects in the <p:row> component is broken. If I don't add elements  as a column in <p:row>, it doesn't get fixed.

For example this code has alignment problems:

`
<p:row>	
		<div class = "ui-grid-col-6 PadRight1Per ResPadRightPer1To0">
		<p:selectOneMenu id = "someDataHere" converter="elemanConverter" required = "true" value = "#{ControllerHere}">
		<f:selectItem itemLabel = "#ValueHere" itemValue = "" />
		<f:selectItems itemLabel = "#{ValueHere}" itemValue = "#{ValueHere}" value = "#{ControllerHere}" var = "?" />
		<p:ajax global = "true" listener = "#{ControllerHere}" process = "@this" update = "@this" />
		</p:selectOneMenu>
		</div> 
		<div class = "ui-grid-col-6 PadLeft1Per ResPadLeftPer1To0">   
		<p:selectOneMenu id = "someDataHere" required = "true" value = "#{ControllerHere}">
		<f:selectItem itemLabel = "#{ValueHere" itemValue = "" />
		<f:selectItems itemLabel = "#{ValueHere}" itemValue = "#{ValueHere}" value = "#{ControllerHere}" var = "?" />
		<p:ajax global = "true" listener = "#{ControllerHere}" process = "@this" update = "@this" />
		</p:selectOneMenu>
		</div> 	
</p:row>
`

After I add <p:column> like that problem solves:

`
<p:row>	
	<p:column>
		<div class = "ui-grid-col-6 PadRight1Per ResPadRightPer1To0">
		<p:selectOneMenu id = "someDataHere" converter="elemanConverter" required = "true" value = "#{ControllerHere}">
		<f:selectItem itemLabel = "#ValueHere" itemValue = "" />
		<f:selectItems itemLabel = "#{ValueHere}" itemValue = "#{ValueHere}" value = "#{ControllerHere}" var = "?" />
		<p:ajax global = "true" listener = "#{ControllerHere}" process = "@this" update = "@this" />
		</p:selectOneMenu>
		</div> 
		<div class = "ui-grid-col-6 PadLeft1Per ResPadLeftPer1To0">   
		<p:selectOneMenu id = "someDataHere" required = "true" value = "#{ControllerHere}">
		<f:selectItem itemLabel = "#{ValueHere" itemValue = "" />
		<f:selectItems itemLabel = "#{ValueHere}" itemValue = "#{ValueHere}" value = "#{ControllerHere}" var = "?" />
		<p:ajax global = "true" listener = "#{ControllerHere}" process = "@this" update = "@this" />
		</p:selectOneMenu>
		</div> 
	<p:column>		
</p:row>
`

But I have 2000 elements like this in my app, Is there any way to fix all of this completely? What should I do?

![Screenshot_6](https://github.com/primefaces/primefaces/assets/33330165/b911c8b6-eab3-4759-bf7a-3e95197be0aa)

**Comment by melloware** [comment, marked as answer]
@goxostar unfortunately NO other than fixing your code.  That original code of `<p:row>` with no `<p:column>` should have never been supported.  Once you are using rows and columns it requires there to be columns inside the row.

So the short answer is you will have to update your code.

**Reply by goxostar** [reply]
thank you, there is nothing to do then I will update all the codes
